frontend/src/pages/modelRegistry/screens/ModelVersionDetails/ModelVersionRegisteredFromLink.tsx

```diff
@@ -0,0 +1,134 @@
+            registeredFromPipelineDetails.project,
+            registeredFromPipelineDetails.runId,
+          )}
+        >
+          {registeredFromPipelineDetails.runName}
+        </Link>
+      ) : (
+        <b>{registeredFromPipelineDetails.runName}</b>
+      );
+
+      const renderProject = registeredPipelineProject
+        ? registeredPipelineProject.metadata.annotations?.['openshift.io/display-name'] ??
+          registeredPipelineProject.metadata.name
+        : registeredFromPipelineDetails.project;
+
+      return (
+        <Flex
+          spaceItems={{ default: 'spaceItemsXs' }}
+          alignItems={{ default: 'alignItemsCenter' }}
+          data-testid="registered-from-pipeline"
+        >
+          <FlexItem data-testid="pipeline-run-link">Run {renderRunLink} in</FlexItem>
+          <FlexItem style={{ display: 'flex' }}>
+            <TypedObjectIcon
+              resourceType={ProjectObjectType.project}
+              style={{ height: 24, width: 24 }}
+            />
+          </FlexItem>
+          <FlexItem>
+            <b>{renderProject}</b>
+          </FlexItem>
+          {!registeredPipelineProject && (
+            <FlexItem>
+              <Popover
+                data-testid="project-access-info-popover"
+                bodyContent={
+                  <Stack hasGutter>
+                    <StackItem>
+                      <Content>
+                        You don&apos;t have access to the source run. To request access, contact
+                        your administrator.
+                      </Content>
+                    </StackItem>
+                    <StackItem>
+                      <Content>Your administrator might be:</Content>
+                      <PopoverListContent
+                        data-testid="pipeline-project-button-popover"
+                        listItems={[...FindAdministratorOptions, 'Your professor (at a school)']}
+                      />
+                    </StackItem>
+                  </Stack>
+                }
+              >
+                <Button
+                  data-testid="project-access-info-button"
+                  icon={<OutlinedQuestionCircleIcon />}
+                  variant="link"
+                  aria-label="more info"
+                />
+              </Popover>
+            </FlexItem>
+          )}
+        </Flex>
+      );
+    }
+
+    if (registeredFromCatalogDetails) {
+      const catalogModelUrl = getCatalogModelDetailsRoute(registeredFromCatalogDetails);
+      return (
+        <Link to={catalogModelUrl} data-testid="registered-from-catalog">
+          <span style={{ fontWeight: 'var(--pf-t--global--font--weight--body--bold)' }}>
+            {registeredFromCatalogDetails.modelName} ({registeredFromCatalogDetails.tag})
+          </span>{' '}
+          in Model catalog
+        </Link>
+      );
+    }
+
+    return null;
+  };
+
+  const content = renderContent();
+  if (!content) {
+    return null;
+  }
```

Comment:
It could have been a little simpler to just have a `let content = null;` at the top and have each of the if cases set it instead of returning, and then you wouldn't need to wrap it in a renderContent function. But that's such a tiny nit, please feel free to ignore it, this is fine.